vis/graphs.js

```diff
@@ -0,0 +1,221 @@
+/* eslint-disable no-undef */
+
+var num_projects = [];
+var y_descriptions = [];
+var type_quarter = [];
+
+d3.json('projects-extended.json').then((res) => {
+    Object.keys(res).forEach(year => {
+        const quarters = Object.keys(res[year]).reverse();
+        const quartersinfo = Object.values(res[year]).reverse();
+        for(let i = 0; i < Object.values(quarters).length; i++){
+            let quarter = year + ' ' + Object.values(quarters)[i];
+            type_quarter.push(quarter);
+            num_projects.push(Object.keys(Object.values(quartersinfo)[i]).length-3);
+
+            let project = Object.values(quartersinfo)[i];
+            project_desc = Object.values(project).filter(x => x.Description);
+            y_descriptions.push(project_desc);
+        }
+    });
+
+    let proj_type_num = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]; // number of projects for each category
+    let proj_type_quarter = [[0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]];
+
+
+    for (quarter in y_descriptions) {
+        for (desc in y_descriptions[quarter]) {
```

Comment:
im guessing this is failing the linter?

you need to declare these variables..

i would prefer if u change this to `Array.forEach` or `Array.map`